src/main/webapp/js/new_simulation.js

```diff
@@ -55,8 +84,10 @@ function setDynamicUpperBound(boundedInputId, boundingInputId) {
  * Verify the following constrains are kept:
  * awakenessCycle <= roundsNum
  * awakenessDuration <= awakenessCycle
+ * all input tags' values are legal
  */
 function setAllDynamicUpperBounds() {
+    enforceRigidBounds();
     setDynamicUpperBound('awakenessCycle', 'roundsNum');
     setDynamicUpperBound('awakenessDuration', 'awakenessCycle');
```

Comment:
Consider renaming those functions s.t either all of them start with 'set' or all of them start with 'enforce'.